typos in volume 2

gabrieltoliveira opened the pull request.

aszasz commented:
Thank you, Gabriel! All good catches! The text changes are sure good for merge @jonasmalacofilho 
I just don't know about the .gitignore, because I wish to keep track of images... I will let Jonas suggest an approach to that

gabrieltoliveira commented:
Sorry, I could not let the .gitignore out of my revision. Please
disconsider it !

On Thu, May 18, 2017 at 11:18 AM, aszasz <notifications@github.com> wrote:

> Thank you, Gabriel! All good catches! The text changes are sure good for
> merge @jonasmalacofilho <https://github.com/jonasmalacofilho>
> I just don't know about the .gitignore, because I wish to keep track of
> images... I will let Jonas suggest an approach to that
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ITDP/the-online-brt-planning-guide/pull/160#issuecomment-302417489>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APH2_TvDRGhMLRkZGgBr3VRRk_cIyKc9ks5r7FMigaJpZM4NfIXU>
> .
>



-- 

Gabriel T. de Oliveira
+55 21 99995.1708

jonasmalacofilho commented:
@gabrieltoliveira  I've rewritten the history for this pull request excluding that first commit (bc416c2).

Watch out if you're still making additions here, since I've changed _your_ fork's history.  You'll need to:

 1. stash any work not yet commited (`git stash`),
 2. fetch your fork (`git fetch <remote>`),
 3. rewrite your local commits on top of what's on the remote (`git rebase <remote>/master`),
 4. and, finally, reapply your stashed changes (`git stash apply`)

where `<remote>` is the name you've given to the remote for `gabrieltoliveira/the-online...`.  You can can check that with `git remote -v`.

---

By the way, here's a pretty good guide on how to perform some common history manipulations in git (like removing commits): https://sethrobertson.github.io/GitFixUm/fixup.html

Also, you might find it easier to work in branches (`git checkout -b <name-for-new-branch>`) in the long run.

Oh, and I'm curious: what made you need  to add the assets to the `.gitignore`?

gabrieltoliveira commented:
Thank you for the extensive feedback. I wanted to clone the repository to
my Desktop, but not have to download all the images or keep track of image
changes. In other words, I wanted just to have the text files so I could
change them easily, without consuming too much hard disk memory.

On Thu, May 18, 2017 at 5:01 PM, Jonas Malaco <notifications@github.com>
wrote:

> @gabrieltoliveira <https://github.com/gabrieltoliveira> I've rewritten
> the history for this pull request excluding that first commit (bc416c2
> <https://github.com/ITDP/the-online-brt-planning-guide/commit/bc416c21f4dd9f1e44b3dde8202f233d022647d3>
> ).
>
> Watch out if you're still making additions here, since I've changed *your*
> fork's history. You'll need to:
>
>    1. stash any work not yet commited (git stash),
>    2. fetch your fork (git fetch <remote>),
>    3. rewrite your local commits on top of what's on the remote (git
>    rebase <remote>/master),
>    4. and, finally, reapply your stashed changes (git stash apply)
>
> where <remote> is the name you've given to the remote for
> gabrieltoliveira/the-online.... You can can check that with git remote -v.
> ------------------------------
>
> By the way, here's a pretty good guide on how to perform some common
> history manipulations in git (like removing commits):
> https://sethrobertson.github.io/GitFixUm/fixup.html
>
> Also, you might find it easier to work in branches (git checkout -b
> <name-for-new-branch>) in the long run.
>
> Oh, and I'm curious: what made you need to add the assets to the
> .gitignore?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ITDP/the-online-brt-planning-guide/pull/160#issuecomment-302524828>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APH2_edRXW2K2NgKa-znoRvTbVkhPgVIks5r7KOTgaJpZM4NfIXU>
> .
>



-- 

Gabriel T. de Oliveira
+55 21 99995.1708

jonasmalacofilho commented:
Oh, great news then!  We use [git-lfs](https://git-lfs.github.com/) in this repository mostly because of the large number of pictures...

If you have `git` installed, but not `git-lfs` (this is the case if get an error from running `git lfs env`), you should be able to work on the repository without having the images; you'll still see `*.jpg` and other image files, but they'll only be small text placeholders.

Still, you can also have `git-lfs` installed but only get the image files you want.  You do this by disabling the automatic behavior, either all the time (`git lfs install --local --skip-smudge`) or on a need-to basis (by setting the `GIT_LFS_SKIP_SMUDGE=1` environment variable); later, you can download all, some or no assets, depending on what you need.

If you want, I can give you some additional pointers...  You'll find most of the documentation for this in the `git-lfs` manual (available from `git lfs help`).

gabrieltoliveira commented:
Jonas, sorry for my question but do you know if this is automatic when I
clone the repository through GitHub Desktop? Because right now, I'm with
all the images in the respective assets folder of each chapter. I do have
an *enforce_lfs *in the root folder of my directory, though.

On Thu, May 18, 2017 at 5:24 PM, Jonas Malaco <notifications@github.com>
wrote:

> Oh, great news then! We use git-lfs <https://git-lfs.github.com/> in this
> repository mostly because of the large number of pictures...
>
> If you have git installed, but not git-lfs (this is the case if get an
> error from running git lfs env), you should be able to work on the
> repository without having the images; you'll still see *.jpg and other
> image files, but they'll only be small text placeholders.
>
> Still, you can also have git-lfs installed but only get the image files
> you want. You do this by disabling the automatic behavior, either all the
> time (git lfs install --local --skip-smudge) or on a need-to basis (by
> setting the GIT_LFS_SKIP_SMUDGE=1 environment variable); later, you can
> download all, some or no assets, depending on what you need.
>
> If you want, I can give you some additional pointers... You'll find most
> of the documentation for this in the git-lfs manual (available from git
> lfs help).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ITDP/the-online-brt-planning-guide/pull/160#issuecomment-302530799>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APH2_bzoKkmi_wg8nYeRUe47BXANhHZnks5r7KjigaJpZM4NfIXU>
> .
>



-- 

Gabriel T. de Oliveira
+55 21 99995.1708

jonasmalacofilho commented:
I think it is, but it should be possible to reconfigure that afterwards.

I can take a closer look on Monday... I'm traveling this week and I don't have access to a Windows machine.  By the way, are you on Windows or Mac?

jonasmalacofilho commented:
Oh, `.enforce_lfs` plays no role in this, oddly enough.  It's something used only by the server to make sure that no images are committed directly, without using LFS (that would be bad beucause it would force everyone to have [all versions of] that image at all times).

jonasmalacofilho commented on guide/13-business-structure/04-competition-within-the-brt-market.manu:
```diff
@@ -16,7 +16,7 @@ until the next tender. If the quality of service is poor, the government has lit
 
-Competition “for the market,” by contrast, traditionally meant ongoing competition for customers for
+Competition “in the market,” by contrast, traditionally meant ongoing competition for customers for
 the same set of services. The traditional downside of competition “for the market” was that firms would
```
I just noticed that @aszasz had mentioned something that got lost during the rebase:

> This is right and I think line 20 should replace expression `"for the market"` by `"in the market"` as well... _for_ line 19 and 20 are discussing the same type of competition.

jonasmalacofilho commented:
@gabrieltoliveira sorry it took me a while, I wanted to test this on completely clean Windows installation...

GitHub Desktop does support LFS and, by default, checks out all assets automatically.

It doesn't come with [exposed] git/git-lfs command-line interfaces [anymore], so you would probably benefit from installing them.  The Windows installation for [git](https://git-scm.com) already comes with LFS and the default options are [most likelly] already what you want/need.

Once you have them installed and set up – `git lfs version` should output something useful on the command prompt you will use and you should have executed `git lfs install` _once_ (you might need to reboot your machine for the prompt to find `git`) – you can place LFS in "manual" mode for this project by issuing `git lfs install --local --skip-smudge` from [a prompt window and a path] within your local repository folder.

After this, no assets will be automatically checked out _anymore_, either from the command line tools or GitHub Desktop.  On the other hand,  you will need to remember to fetch the figures manually – with `git lfs pull` or other `git lfs ...` commands – when you do need them.

---

[Advanced:] Additionally, if you want to remove all assets already checked out:

 - 1. make sure you have commited and pushed all changes and new files
 - 2. delete all files and folders except for the `.git` folder
 - 3. delete the `.git/lfs` folder
 - 4. run `git reset --hard`

---

Here are two tips for quickly opening a prompt window (also know as a "shell") in the correct path:

 - from GitHub Desktop, go to Repository -> Open command prompt
 - from the Windows Explorer, <kbd>SHIFT</kbd> + right-click the whitespace between the file icons -> Open command window here

jonasmalacofilho commented:
Note: there are also the `lfs.fetchinclude`/`lfs.fetchexclude` settings, and `git reset --hard` should even clean unwanted files by itself, provided these settings [and some other conditions](https://github.com/git-lfs/git-lfs/issues/1189#issuecomment-215477888).  But I wasn't able to get this to work on Windows...
